pkg/lib/sms/accessyou/smsclient.go

```diff
@@ -76,9 +77,36 @@ func (n *AccessYouClient) Send(ctx context.Context, options *smsclient.SendOptio
 	}
 
 	// Failed case.
-	return nil, &smsclient.SendResultError{
+	return nil, n.makeError(sendSMSResponse.Status, dumpedResponse)
+}
+
+func (n *AccessYouClient) makeError(
+	msgStatus string,
+	dumpedResponse []byte,
+) *smsclient.SendResultError {
+	err := &smsclient.SendResultError{
 		DumpedResponse: dumpedResponse,
 	}
+
+	// See https://www.accessyou.com/smsapi.pdf
+	switch msgStatus {
+	case "108":
+		fallthrough
+	case "110":
+		err.Code = api.CodeInvalidPhoneNumber
+		err.ErrorDetail = msgStatus
+	case "105":
+		err.Code = api.CodeAuthenticationFailed
+		err.ErrorDetail = msgStatus
+	case "106":
+		fallthrough
+	case "107":
+	case "too_many_login_failure":
+		err.Code = api.CodeAuthorizationFailed
```

Comment:
I think this should be treated as CodeAuthenticationFailed

Reply:
According to the doc, this error code means the account is suspended, so it is a authorization error. It will still persist even you fixed the credentials.